js/battledata.js

```diff
@@ -330,8 +330,41 @@ var Tools = {
 				options.hidestrikethrough ? '$1' : '<s>$1</s>');
 		// linking of URIs
 		if (!options.hidelinks) {
-			str = str.replace(/https?\:\/\/[a-z0-9-.]+(?:\:[0-9]+)?(?:\/(?:[^\s]*[^\s?.,])?)?|[a-z0-9.]+\@[a-z0-9.]+\.[a-z0-9]{2,3}|(?:[a-z0-9](?:[a-z0-9-\.]*[a-z0-9])?\.(?:com|org|net|edu|us|jp)(?:\:[0-9]+)?|qmark\.tk)(?:(?:\/(?:[^\s]*[^\s?.,])?)?)\b/ig, function(uri) {
-				if (/[a-z0-9.]+\@[a-z0-9.]+\.[a-z0-9]{2,3}/ig.test(uri)) {
+			var domainRegex = '[a-z0-9\\-]+(?:[.][a-z0-9\\-]+)*';
+			var parenthesisRegex = '[(][^\\s()<>]*[)]';
+			var linkRegex = new RegExp(
+				'\\b' +
+				'(?:' +
+					'(?:' +
+						// When using www. or http://, allow any-length TLD (like .museum)
+						'(?:https?://|www[.])' + domainRegex +
+						'|' + domainRegex + '[.]' +
+							// Allow a common TLD, or any 2-3 letter TLD followed by : or /
+							'(?:com?|org|net|edu|info|us|jp|[a-z]{2,3}(?=[:/]))' +
+					')' +
+					'(?:[:][0-9]+)?' +
+					'\\b' +
+					'(?:' +
+						'/' +
+						'(?:' +
+							'(?:' +
+								'[^\\s()<>]' +
+								'|' + parenthesisRegex +
+							')*' +
+							// URL cannot end with punctation characters
```

Comment:
Well, sure they can (technically), but generally speaking it's odd to find an URL ending with a period, for example. While it's fine to allow punctuation at end if it's let's say URL field on a website (no reason to block it here), in this case I parse plain text, which is far more tricky than a simple URL field. If you see a period at end of URL, chances are it's not a part of URL, but rather a period at end of sentence.